### DISCUSSION TITLE
Official docker image

### DISCUSSION THREAD

**Discussion by tdrz** [baresip/baresip#2003] [Ideas]
Have you guys considered maintaining an "official" docker image? 

I think it would be very useful to have it in [docker hub](https://hub.docker.com). Then with every release, an automated pipeline would create and push a new image, possibly with multiple architectures, to docker hub.

I can help if this is of interest!

**Comment by pataquets** [comment]
Having Docker images availables is super useful for quickly testing from almost any platform without the need to install buildtools and compile. It's also a very welcoming workflow for wanting-to-be contributors, since it takes out all the fuss of deps and build tools from it.
Also, as a distribution/packaging method, it's quite easy and also very cross-platform.

**Comment by sreimers** [comment]
A while ago I created a docker repo, just cleaned up a bit and updated to current versions: https://github.com/baresip/docker
PRs are welcome. 

Instead of docker hub it uses [github packages](https://github.com/features/packages). This way we can directly push images and have no separate account to manage. I just pinned the docker repo to give this more visibility.

**Comment by pataquets** [comment]
Thanks for sharing your image, @sreimers. I will keep it handy in case I need it.
Anyway, I think having an "official" `Dockerfile` in the repo would be beneficial for quickly compiling BareSIP and in other cases to quickly bootstrap contributors. It would be possible to quickly build a BareSIP Docker image locally to run it afterwards by issuing just a one-liner:
```console
$ docker build -t baresip https://github.com/baresip/baresip.git#master
``` 
This can also be used from Docker Compose manifests. It could even work for different branches/tags for testing features.
@alfredh Just to know your stance, would you be open to host on the repo a proper `Dockerfile` which compiled BareSIP from scratch? Just to make it visible to possibly interested parties which might even step up to do it.

EDIT: Docker images would also serve as methods for packaging BareSIP + custom configs for specific use cases in self-contained artifacts (e.g. #2424).
